### DISCUSSION TITLE
How do you move a window to a new container?

### DISCUSSION THREAD

**Discussion by NullVoxPopuli** [nikitabobko/AeroSpace#798] [questions-and-answers]
I see the recommendation is to use join-width, and collapsing containers is not possible with `split`.

I like collapsing containers, but I don't see a way to move windows out of an existing container.

For example, If I have:
```
|---------------------|
|    2     |     1    |
|          |          |
|---------------------|
|    3     |     4    |
|---------------------|
```
and I want to move window `1` to the right of everything, it doesn't seem possible

This is what I'm trying to do:
```
|----------------------|
|    2     |  4  | 1   |
|          |     |     |
|----------|     |     |
|    3     |     |     |
|----------------------|
```

but I can't figure it out with join-with. 
Why is join-with and split mutually exclusive with container collapsing?

in whatever situation I'm in doing join-with up/down on 1/4 seems to not do anything either

**Comment by gldtn** [comment]
My tip is to open 4 terminal OS windows. Go to each of them or the two you are trying to re-arrange. 

Run `aerospace list-windows --focused `, this will give you the id of the currently focused window. You might need this id in the case you need to move that specific window to join with the other. You can also run `aerospace list-windows --workspace focused` to retrieve all the open window's id at once. 

You can then play around by running `aerospace join-with up` or whatever direction you need, it should output if there is no window available on that direction. If you are trying to move up for example, and aerospace tells you there is no window window available, you can try bringing that specific window down by using the window id itself with `aerospace join-with down --window-id <window-id>` .

Read about normalization here: https://nikitabobko.github.io/AeroSpace/guide#normalization
Also see this issue: https://github.com/nikitabobko/AeroSpace/issues/260

Hope it helps!

**Reply by NullVoxPopuli** [reply]
![image](https://github.com/user-attachments/assets/3779f828-5606-401c-83a3-e33ce4d48c25)

Nothing happened 
![image](https://github.com/user-attachments/assets/1063104b-3101-4a70-88a5-0f3846deb931)

(same for `aerospace join-with left --window-id 14723`)

It didn't say there was no window available.

> Read about normalization here: https://nikitabobko.github.io/AeroSpace/guide#normalization

thanks! so it seems this layout is sort of what I'm trying to do -- create a stack of 3 windows
```
# According to the second normalization, such layout isn’t possible:
h_tiles
├── window 1
└── h_tiles
    ├── window 2
    └── window 3
```
though, I would expect it to be something like this
```
h_tiles
├── 
└── h_tiles
    |-- window 1
    ├── window 2
    └── window 3
```
which would be flattened to
```
h_tiles
|-- window 1
├── window 2
└── window 3
```

**Reply by gldtn** [reply]
If I understood you right, It should be fairly simple.. I'm currently trying to get a script to manipulate windows the way I want, so I decided to make a small video for you. If your window is not moving at all, something else might be going on. Although from your screenshots it seems to be working for you to get your browser to the left below the first alacritty window. `No windows in specified direction` should appear only if there is actually no other window opened in the direction you are trying to move to.

Try to replicate what I did in the small video below. Hope it helps!

https://github.com/user-attachments/assets/33b90ba2-69dd-4afe-af50-2abfc264a739
